doc/api/n-api.md

```diff
@@ -3134,6 +3134,57 @@ overhead in creating/storing strings with this method.
 The JavaScript `string` type is described in
 [Section 6.1.4][] of the ECMAScript Language Specification.
 
+### `node_api_create_property_key_utf8`
+
+<!-- YAML
+added:
+  - v22.1.0
+-->
+
+#### `napi_status node_api_create_property_key_utf8(napi_env env, const char* utf8name, size_t length, napi_value* result)`
+
+* `env`: `napi_env`
+* `utf8name`: `const char*`
+* `length`: `size_t`
+* `result`: `napi_value*`
+
+This function creates a property key from a UTF-8 encoded string.
+
+```c
+napi_value key;
+napi_status status = node_api_create_property_key_utf8(env, "myKey", strlen("myKey"), &key);
```

Comment:
For a zero-terminating string we recommend using `NAPI_AUTO_LENGTH` instead. We call the `strlen` internally in such case. The explicit length is only required if the string may have `\0` inside or we want to use a substring. These two parameters are in some sense similar to the `std::string_view` behavior.

Reply:
many thanks I tried to apply as you said